install/scripts/craycli_init.py

```diff
@@ -0,0 +1,753 @@
+        # check the input
+        if username == "":
+            LOGGER.error("Attempting to delete empty username")
+            return
+
+        # get the user id of the desired user
+        userId = self.get_cli_user_id(username)
+        if userId == "":
+            LOGGER.debug(f"Specified user {username} does not exist in keycloak")
+            return
+
+        # user exists, so delete this user from keycloak
+        url = f"{self.keycloak_base}/admin/realms/{self.SHASTA_REALM_NAME}/users/{userId}"
+        response = self._kc_master_admin_client.delete(url)
+        response.raise_for_status()
+        LOGGER.info("Deleted user %r", username)
+
+    # get the keycloak user id for a given username
+    def get_cli_user_id(self, username):
+        LOGGER.debug(f"Looking for id of user: {username}")
+        # check the input
+        if username == "":
+            LOGGER.error("Attempting to find the user id of an empty username")
+            return ""
+
+        # search for the users that match the expected user
+        url = (f"{self.keycloak_base}/admin/realms"
+               f"/{self.SHASTA_REALM_NAME}/users/?username={username}")
+        response = self._kc_master_admin_client.get(url)
+        response.raise_for_status()
+
+        # parse through the responses - returns a list of json objects
+        user_id = ""
+        items = response.json()
+        for item in items:
+            if item["username"] == username:
+                user_id = item["id"]
+
+        # report what we found
+        LOGGER.debug(f"Keycloak user id: {user_id}")
+        return user_id
+
+    # get the keycloak user id for a given username
+    def get_grp_id(self, groupname):
+        # check the input
+        if groupname == "":
+            LOGGER.error("Attempting to find the group id of an empty groupname")
+            return ""
+
+        # search for the users that match the expected user
+        url = (f"{self.keycloak_base}/admin/realms/{self.SHASTA_REALM_NAME}/"
+                f"groups?search={groupname}")
+        response = self._kc_master_admin_client.get(url)
+        response.raise_for_status()
+
+        # parse through the responses - returns a list of json objects
+        group_id = ""
+        items = response.json()
+        for item in items:
+            if item["name"] == groupname:
+                group_id = item["id"]
+
+        # report what we found
+        LOGGER.debug(f"Group id: {group_id}")
+        return group_id
+
+def read_keycloak_master_admin_secrets(k8sClientApi):
+    # read the secrets from the k8s client
+    try:
+        sec = k8sClientApi.read_namespaced_secret("keycloak-master-admin-auth",
+                "services").data
+        return {
+            'client_id': base64.b64decode(sec['client-id']),
+            'user': base64.b64decode(sec['user']),
+            'password': base64.b64decode(sec['password'])
+        }
+    except client.exceptions.ApiException:
```

Comment:
You can define the exception as a variable.
except client.exceptions.ApiException as error:

Then you can reference the 'error' variable in your error output.

Reply:
Done.